src/openms/include/OpenMS/FILTERING/SMOOTHING/SavitzkyGolayFilter.h

```diff
@@ -175,7 +176,7 @@ namespace OpenMS
         }
 
         out_it->setPosition(first->getPosition());
-        out_it->setIntensity(std::max(0.0, help));
+        out_it->setIntensity(std::max<IntensityType>(0, help));
```

Comment:
Wait. Why the different usage here?

Reply:
compared to before

Reply:
no particular reason. But needs reverting anyway. No idea why it breaks... we have IEEE math .. but not on MacOS :)